Assist tutorial on setting up phone to control HA

c0ffeeca7 opened the pull request:
## Proposed change
<!-- 
    Describe the big picture of your changes here to communicate to the
    maintainers why we should accept this pull request. If it fixes a bug
    or resolves a feature request, be sure to link to that issue in the 
    additional information section.
-->
- Add new folder projects/private-voice-assistant
- Add tutorial on setting up a classic analog phone to control Home Assistant 
  -  Instructions on setting up Grandstream
  - Instructions on setting up Voice over IP integration 
  - Instructions on setting up voice assistant (basic setup using default settings, not including Alexa or Google) 
  - Troubleshooting information
- Link to tutorial from front page
- Link to tutorial from documentation page, asides
- some content and screenshots are copied from @synesthesiam, https://github.com/home-assistant/home-assistant.io/pull/27085
- video courtesy of @TheFes 


## Type of change
<!--
    What types of changes does your PR introduce to our documentation/website?
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR.
-->

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [x] Added documentation for a new integration others are adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information
<!--
    Details are important, and help maintainers processing your PR.
    Please be sure to fill out additional details, if applicable.
-->

- Link to parent pull request in the codebase: 
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist
<!--
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR. If you're unsure about any of them, don't hesitate to ask.
    We're here to help! This is simply a reminder of what we are going to look
    for before merging your code.
-->

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

c0ffeeca7 committed:
Assist tutorial on setting up phone to control HA

- Add new folder projects/private-voice-assistant
- Add tutorial on setting up a classic analog phone to control Home Assistant
-- Instructions on setting up Grandstream
-- Instructions on setting up Voice over IP integration
-- Instructions on setting up voice assistant (basic setup using default settings, not including Alexa or Google)
-- Troubleshooting information
- Link to tutorial from front page
- Link to tutorial from documentation page, asides

c0ffeeca7 committed:
VoIP tutorial: add step to activate assistant

c0ffeeca7 committed:
Remove unused images

c0ffeeca7 committed:
Remove unused ESPHome images

c0ffeeca7 committed:
VoIP tutorial: fix grammar: set up vs setup

c0ffeeca7 committed:
Merge branch 'next' into setup-assist-voip-tutorial

c0ffeeca7 committed:
Merge branch 'next' into setup-assist-voip-tutorial

balloob commented on source/projects/private-voice-assistant/voice_remote_classic_phone.markdown:
```diff
@@ -0,0 +1,154 @@
+---
```
The docs sidebar is not shown on this page. We have an aside that checks the URL to see which sidebar to load, it should include /projects/

frenck commented on source/projects/private-voice-assistant/voice_remote_classic_phone.markdown:
```diff
@@ -0,0 +1,154 @@
+
+This tutorial shows how to set up an analog phone to control Home Assistant with your voice.
+The phone is used to implement a push-to-talk mechanism. 
+In this way, the beginning and end of a voice stream are clearly defined.
```
```suggestion
This way, the beginning and end of a voice stream are clearly defined.
```

synesthesiam commented on source/projects/private-voice-assistant/voice_remote_classic_phone.markdown:
```diff
@@ -0,0 +1,154 @@
+---
+
+This tutorial shows how to set up an analog phone to control Home Assistant with your voice.
+The phone is used to implement a push-to-talk mechanism. 
```
I wouldn't quite call it a "push to talk" mechanism since you can speak more than one voice command without hanging up. Maybe something like "This allows you to have a private conversation with your smart home that only starts when you pick up the phone, and stops listening the moment you hang up."

frenck commented on source/projects/private-voice-assistant/voice_remote_classic_phone.markdown:
```diff
@@ -0,0 +1,154 @@
+
+## Setting up the phone in Home Assistant
+
+1. In Home Assistant, go to {% my integrations title="**Settings** > **Devices & Services**" %} and select **Add integration**.
```
Should we add a my link to add the integration instead directly? (or additional?)

balloob replied:
Use a My link to link to a specific integration, similar to integration docs. See for example WLED https://next.home-assistant.io/integrations/wled/

balloob commented on source/projects/private-voice-assistant/voice_remote_classic_phone.markdown:
```diff
@@ -0,0 +1,154 @@
+  [Grandstream HT801](https://www.grandstream.com/products/gateways-and-atas/analog-telephone-adaptors/product/ht801) or 
+  [HT802](https://www.grandstream.com/products/gateways-and-atas/analog-telephone-adaptors/product/ht802)
+  * includes a 5&nbsp;V power adapter and an Ethernet cable
+* RJ11 cable to connect the phone to the Grandstream
```
```suggestion
* RJ11 phone cable to connect the phone to the Grandstream
```

frenck commented on source/projects/private-voice-assistant/voice_remote_classic_phone.markdown:
```diff
@@ -0,0 +1,154 @@
+   * Say a [supported voice command](/docs/assist/builtin_sentences/). For example, *Turn off the light in the kitchen*.
+   * You can also ask a question, such as
+     *  *Is the front door locked?*
+     *  *Which lights are on in the living room?*.
```
```suggestion
     *  *Which lights are on in the living room?*
```

synesthesiam commented on source/projects/private-voice-assistant/voice_remote_classic_phone.markdown:
```diff
@@ -0,0 +1,154 @@
+   * \*47192\*168\*1\*100\*5060
+   ![Define IP for autodial](/images/assist/grandstream_autodial.png)
+   * Note: instead of 192\*168\*1\*100\*, enter the IP address of your Home Assistant instance.
+   * If you have a Grandstream HT802, there are two ports to configure. Repeat this step for the second port.
```
The second port can be configured to call a different Home Assistant device, but it might be worth noting that a single Home Assistant device can't distinguish between the two ports.

balloob replied:
I think that we should remove all mention of the HT802. It makes the tutorial more complicated than necessary

c0ffeeca7 replied:
Removed the references to the HT802

frenck commented on source/projects/private-voice-assistant/voice_remote_classic_phone.markdown:
```diff
@@ -0,0 +1,154 @@
+
+## Troubleshooting
+
+Things are not working as expected? This section lists a few steps that may help you troubleshoot issues.
```
```suggestion
Are things not working as expected? This section lists a few steps that may help you troubleshoot issues.
```

synesthesiam commented on source/projects/private-voice-assistant/voice_remote_classic_phone.markdown:
```diff
@@ -0,0 +1,154 @@
+1. Start up the Grandstream.
+   * Connect the power supply.
+   * Once the Grandstream has booted, the two LEDs for power and Ethernet light up solid blue. The phone LED won't light up.
+1. Log onto your router and identify the IP address of the Grandstream.
```
You can also get the phone to tell you its IP address! https://documentation.grandstream.com/knowledge-base/ht801-quick-installation-guide/#obtaining-the-ip-address

c0ffeeca7 replied:
That's cool!

frenck commented on source/projects/private-voice-assistant/voice_remote_classic_phone.markdown:
```diff
@@ -0,0 +1,154 @@
+
+## About the analog phone
+
+You can use any analog landline phone, ideally one that already has an RJ11 socket.
```
```suggestion
You can use any analog landline phone with an RJ11 socket.
```

balloob replied:
I would drop the "Ideally part". The RJ11 socket is necessary to connect it to the grandstream

frenck commented on source/projects/private-voice-assistant/voice_remote_classic_phone.markdown:
```diff
@@ -0,0 +1,154 @@
+## About the analog phone
+
+You can use any analog landline phone, ideally one that already has an RJ11 socket.
+During creation of this tutorial this classic phone from 1953 was used:
```
```suggestion
During the creation of this tutorial, this classic phone from 1953 was used:
```

frenck commented on source/projects/private-voice-assistant/voice_remote_classic_phone.markdown:
```diff
@@ -0,0 +1,154 @@
+## FAQ
+
+**Which languages are supported?**
+[These](https://developers.home-assistant.io/docs/voice/intent-recognition/supported-languages/) are the languages supported by Home Assistant.
```
I think just linking `this` or `these` words in a sentence is generally a bit unclear in intention. Especially in this case, we can add a little more context to the link by linking the sentence.

```suggestion
[These are the languages supported by Home Assistant](https://developers.home-assistant.io/docs/voice/intent-recognition/supported-languages/).
```

c0ffeeca7 replied:
Very good point. I'm almost a bit surprised/shocked I did that...

synesthesiam commented on source/projects/private-voice-assistant/voice_remote_classic_phone.markdown:
```diff
@@ -0,0 +1,154 @@
+
+## Setting up a Home Assistant voice assistant
+
+1. Set up text-to-speech in Home Assistant Cloud.
```
Do we need to add some text to say that HA cloud is required here? Maybe also a link to a tutorial for setting up a local voice assistant when we have it.

balloob replied:
I think that we can remove this whole section. Instead, at requirements we can specify you need to have Home Assistant Cloud or manually configured an Assist Pipeline.

synesthesiam commented on source/projects/private-voice-assistant/voice_remote_classic_phone.markdown:
```diff
@@ -0,0 +1,154 @@
+## Setting up a Home Assistant voice assistant
+
+1. Set up text-to-speech in Home Assistant Cloud.
+   * Once the pipeline is further developed, the cloud won't be necessary. It will be possible to process speech locally. 
```
Not necessary, but it will be a much better experience (especially for users on a Raspberry Pi).

c0ffeeca7 replied:
Rephrased. Let me know what you think.

synesthesiam commented on source/projects/private-voice-assistant/voice_remote_classic_phone.markdown:
```diff
@@ -0,0 +1,154 @@
+1. Go to **Settings** > **Voice assistants** and select **Add assistant**.
+   * Enter a name. You can pick any name that is meaningful to you.
+   ![Enter a name for your voice assistant](/images/assist/assistant-give-name.png)
+   * Select a **Conversation agent**. This is the software that processes your voice stream. 
```
The conversation agent processes the voice commands, but not the audio directly. I don't know if that matters here, but it might be unclear how its different from speech-to-text.

c0ffeeca7 replied:
ah, I see. I was confused about that.

balloob commented on source/projects/private-voice-assistant/voice_remote_classic_phone.markdown:
```diff
@@ -0,0 +1,154 @@
+   * Calls from new devices are blocked by default since voice commands could be used to control sensitive devices, such as locks and garage doors.
+   * In the **Voice over IP** integration, select the **device** link.
+   * To allow this phone to control your smart home, under **Configuration**, enable **Allow calls**.   
+   ![Voice over IP integration - allow calls](/images/assist/voip_configuration.png) 
```
I wonder if we should show a whole device info page screenshot, so it's clear where the configuration box is.

c0ffeeca7 replied:
Yes, I see your issues there. I started out with the entire dialog. But I found it was too big and split it in subsections... For the moment, I split them 50/50. Let me know what you think.

synesthesiam commented on source/projects/private-voice-assistant/voice_remote_classic_phone.markdown:
```diff
@@ -0,0 +1,154 @@
+      ![Select the language you want to speak](/images/assist/assistant-conversation-agent.png)
+   * In the **Speech-to-text** and **Text-to-speech** drop-down menus, select an assistant. For this tutorial, select **Home Assistant Cloud**.
+   * Under **Text-to-speech**, select the **voice** of the assistant. For this tutorial, use the default voice.
+   * Select **Save**.
```
Save -> Create

synesthesiam commented on source/projects/private-voice-assistant/voice_remote_classic_phone.markdown:
```diff
@@ -0,0 +1,154 @@
+   * Under **Configuration**, in the **Assist pipeline** menu, select your newly created assistant from the list.   
+      ![Activate your newly created assistant](/images/assist/voip-select-pipline.png)
+1. Congratulations! You set up your analog phone to work with Home Assistant. Now pick up the phone and control your device. 
+   * Say a [supported voice command](/docs/assist/builtin_sentences/). For example, *Turn off the light in the kitchen*.
```
Should we mention that this depends on their area/entity names and aliases, or is that obvious?

c0ffeeca7 replied:
Good point. you would think it's obvious. :-) But I failed at it the first time I tested this.

balloob commented on source/projects/private-voice-assistant/voice_remote_classic_phone.markdown:
```diff
@@ -0,0 +1,154 @@
+   ![Voice over IP integration - allow calls](/images/assist/voip_configuration.png) 
+1. Your phone is set up in Home Assistant. The next step is to set up a voice assistant.
+
+## Setting up a Home Assistant voice assistant
```
We will automatically create an assist pipeline now, users don't need to set up.

I think we should link to the `assist_pipeline` integration, which describes how to manage pipelines.

balloob commented on source/projects/private-voice-assistant/voice_remote_classic_phone.markdown:
```diff
@@ -0,0 +1,154 @@
+   * Under **Text-to-speech**, select the **voice** of the assistant. For this tutorial, use the default voice.
+   * Select **Save**.
+   ![Enter a name for your voice assistant](/images/assist/assistant-stt-tts-define.png)
+1. To be able to control your devices over a voice command, you must expose your entities to Assist:
```
This should probably be moved to a specific documentation page about exposing entities. As it is relevant for any Assist Pipeline, but also Google/Alexa.

balloob commented on source/projects/private-voice-assistant/voice_remote_classic_phone.markdown:
```diff
@@ -0,0 +1,154 @@
+
+It also means that Home Assistant only listens when you want it to.
+
+<lite-youtube videoid="bPDyDFUHDCk" videotitle="Using an analog phone to control Home Assistant"></lite-youtube>
```
Discussed with TheFes, we're moving the video to the HA channel. 
```suggestion
<lite-youtube videoid="0YJzLIMrnGk" videotitle="Using an analog phone to control Home Assistant"></lite-youtube>
```

balloob commented on source/projects/private-voice-assistant/voice_remote_classic_phone.markdown:
```diff
@@ -0,0 +1,154 @@
+---
+title: "Setting up an analog phone to control Home Assistant"
+---
```
Rename this file to `index.markdown` so the url becomes `/projects/private-voice-assistant`

balloob commented on source/projects/private-voice-assistant/voice_remote_classic_phone.markdown:
```diff
@@ -0,0 +1,154 @@
+In this way, the beginning and end of a voice stream are clearly defined.
+This reduces the complexity of voice processing. 
+
+It also means that Home Assistant only listens when you want it to.
```
```suggestion
This tutorial will guide you to turn your old landline phone into the
world's most private voice assistant. Pick up the phone to talk to
your smart home and issue commands and get responses.
```

balloob commented on source/projects/private-voice-assistant/voice_remote_classic_phone.markdown:
```diff
@@ -0,0 +1,154 @@
+  [HT802](https://www.grandstream.com/products/gateways-and-atas/analog-telephone-adaptors/product/ht802)
+  * includes a 5&nbsp;V power adapter and an Ethernet cable
+* RJ11 cable to connect the phone to the Grandstream
+
```
```suggestion
* [Home Assistant Cloud](https://www.nabucasa.com) or a manually configured [Assist Pipeline](/integrations/assist_pipeline)

```

balloob commented on source/projects/private-voice-assistant/voice_remote_classic_phone.markdown:
```diff
@@ -0,0 +1,154 @@
+
+Use this procedure to test if the phrase itself works, without using the voice.
+
+1. Go to the **Developer Tools** and open the **Services** tab.
```
This shouldn't be done here. When debugging a pipeline, you can also send in new text/audio.

balloob replied:
![CleanShot 2023-04-24 at 11 38 39](https://user-images.githubusercontent.com/1444314/234046614-884fcac3-6739-4881-92c9-01356e055e80.png)

balloob commented on source/projects/private-voice-assistant/voice_remote_classic_phone.markdown:
```diff
@@ -0,0 +1,154 @@
+1. At the top of the screen, from the dropdown menu, select the run you are interested in.
+![Debug speech-to-text](/images/assist/assistant-debug-02.png)
+
+### View the logs
```
Is this necessary? All necessary info should be in the pipeline debug info.

c0ffeeca7 replied:
Removed

c0ffeeca7 committed:
Apply suggestions from code review

Co-authored-by: Paulus Schoutsen <balloob@gmail.com>
Co-authored-by: Franck Nijhof <git@frenck.dev>

c0ffeeca7 committed:
Apply suggestions from code review

Co-authored-by: Franck Nijhof <git@frenck.dev>

c0ffeeca7 committed:
Review feedback: update images

c0ffeeca7 committed:
Review feedback: fix sidebar

c0ffeeca7 committed:
Review feedback: rename file to index

c0ffeeca7 committed:
Review comments: remove info on how to view logs

c0ffeeca7 committed:
Implement review feedback

c0ffeeca7 committed:
Review feedback: update troubleshooting procedures

c0ffeeca7 committed:
Merge branch 'next' into setup-assist-voip-tutorial

c0ffeeca7 committed:
Implement review comments: Remove section on setting up assistant

- There will be one setup by default.

c0ffeeca7 committed:
Move generic troubleshooting into separate section

- so that it can be a link target for other pages
- Pointing to the VoIP tutorial for troubleshooting is less ideal,
- as it is more than troubelshooting
- and includes VoIP specific troubleshooting

c0ffeeca7 committed:
Remove FAQ

- It contained a reference to the list of supported languages.
- It is no longer required to check if one's languages is supported,
	 as the UI only proposes supported languages.

c0ffeeca7 committed:
Merge branch 'next' into setup-assist-voip-tutorial

c0ffeeca7 committed:
Merge branch 'next' into setup-assist-voip-tutorial

c0ffeeca7 committed:
add link to custom sentences

-add name of phone model used in video by @TheFes

c0ffeeca7 committed:
Merge branch 'next' into setup-assist-voip-tutorial

balloob commented on source/projects/private-voice-assistant/index.markdown:
```diff
@@ -0,0 +1,94 @@
+
+## Required material
+
+* Latest version of Home Assistant installed on your Home Assistant device
```
```suggestion
* Home Assistant 2023.5 or later
```

c0ffeeca7 committed:
Merge branch 'next' into setup-assist-voip-tutorial

c0ffeeca7 committed:
Implement review feedback

c0ffeeca7 committed:
Implement review feedback. Missed that one

c0ffeeca7 committed:
Fix broken links

c0ffeeca7 committed:
Fix title

c0ffeeca7 committed:
Change troubleshooting file name

c0ffeeca7 committed:
Update screenshot

c0ffeeca7 committed:
Fix title in aside

c0ffeeca7 committed:
Add interview with Mario

c0ffeeca7 committed:
Fix typo

c0ffeeca7 committed:
Merge branch 'next' into setup-assist-voip-tutorial
